src/index.ts

```diff
@@ -22,108 +26,65 @@ async function callElevioApi(path: string, key: string, token: string) {
-  console.log('Finished processing all articles');
-  await mavenAgi.knowledge.finalizeKnowledgeBaseVersion(knowledgeBaseId);
-}
-
-export default {
-  async preInstall({ settings }) {
+const { preInstall, postInstall, knowledgeBaseRefreshed } = new MavenAGIKnowledgeBaseIntegration({
+  preInstallTest: async ({ settings }) => {
     // Make sure the elevio auth token works
     await callElevioApi('/categories', settings.key, settings.token);
   },
+  generatePaginatedResults: async function* ({ settings }) {
+    let page = 1;
+    let hasMorePages = true;
+
+    while (hasMorePages) {
+      const articlesResponse = await callElevioApi(
+        `/articles?status=published&page=${page}`,
+        settings.key,
+        settings.token
+      );
 
-  async postInstall({ organizationId, agentId, settings }) {
-    const mavenAgi = new MavenAGIClient({ organizationId, agentId });
+      const articles = [];
+
+      for (const doc of articlesResponse.articles) {
+        const fullElevioDoc = await callElevioApi(
+          `/articles/${doc.id}`,
+          settings.key,
+          settings.token
+        );
+        const englishTranslation = fullElevioDoc.article.translations.find(
+          (translation) =>
+            translation.language_id === 'en' ||
+            translation.language_id === 'en-us'
+        );
+
+        if (englishTranslation) {
+          articles.push({
+            id: `${doc.id}`,
+            title: englishTranslation.title,
+            content: englishTranslation.body,
+          });
+        } else {
+          console.warn(`No English translation found for article ID: ${doc.id}`);
+        }
+      }
 
-    // Make one maven knowledge base for elevio
-    await mavenAgi.knowledge.createOrUpdateKnowledgeBase({
-      name: 'Elevio',
-      type: MavenAGI.KnowledgeBaseType.Api,
-      knowledgeBaseId: { referenceId: 'elevio' },
-    });
-    await refreshDocumentsFromElevio(
-      mavenAgi,
-      settings.key,
-      settings.token,
-      'elevio'
-    );
-  },
+      yield articles;
 
-  async knowledgeBaseRefreshed({
-    organizationId,
-    agentId,
-    knowledgeBaseId,
-    settings,
-  }) {
-    console.log('Refresh request for ' + knowledgeBaseId.referenceId);
-    const mavenAgi = new MavenAGIClient({ organizationId, agentId });
+      console.log('Finished processing page ' + page + '/' + articlesResponse.total_pages);
+      hasMorePages = page < articlesResponse.total_pages;  
+      page++;
+    }
 
-    // If we get a refresh request, create a new version for the knowledge base and add documents
-    await refreshDocumentsFromElevio(
-      mavenAgi,
-      settings.key,
-      settings.token,
-      knowledgeBaseId.referenceId
-    );
+    console.log('Finished processing all articles');
   },
+  convertDataToMD: (documentBodyPayload: string) => {
+    const turndownService = new TurndownService();
+    turndownService.use(gfm);
+    return turndownService.turndown(documentBodyPayload);

```

Comment:
you don't need turndown here. the api accepts html.
would prefer to pull this logic out of clients and keep it in platform - with the goal of clients being as simple as possible

Reply:
Igor was saying it's better to ship in MD

Reply:
@igorien2k was working on some logic to also convert all sorts of formats into MD

Reply:
Just published v1.0.0 of the library (contains breaking changes).

https://www.npmjs.com/package/@mavenagi/knowledge-base-integration

This allows you to choose `HTML|MD` content. Defaults to HTML and passthrough with examples in the README.

Reply:
platform wise this is a very different code path than what igor was working with - cause this isn't a crawl. 
we basically do turndown for you on the backend - no llms involved here

Reply:
oh and if you run into a different content type that is not html or markdown - definitely let us know! im sure we can find a library to parse it and turn it into markdown in the backend. 

the llm is only needed to clean up content which is nondeterministically messy - and generally speaking that shouldn't be necessary with any of these connectors as they are all first class apis. only crawls are a disaster really